mzLib/Omics/BioPolymer/SequenceVariation.cs

```diff
@@ -1,167 +1,857 @@
+
+        /// <summary>Free-form description (may aggregate provenance / sample info).</summary>
+        public string Description { get; }
+
+        /// <summary>Optional multi-sample VCF record describing the variant (can be null or collapsed).</summary>
+        public VariantCallFormat? VariantCallFormatData { get; }
+
         /// <summary>
-        /// End position of original sequence to be replaced
+        /// Variant-specific modifications keyed by 1-based residue positions in the sequence AFTER variation application.
+        /// Positions are validated in <see cref="AreValid"/> against the altered span (<see cref="VariantSequence"/>).
         /// </summary>
-        public int OneBasedEndPosition { get; }
+        public Dictionary<int, List<Modification>> OneBasedModifications { get; }
 
         /// <summary>
-        /// Original sequence information (optional)
+        /// Unified annotation text for free-form searching/classification.
+        /// Prefers the raw VCF line if available, otherwise the free-form Description.
         /// </summary>
-        public string OriginalSequence { get; }
+        public string SearchableAnnotation => VariantCallFormatData?.Description ?? Description ?? string.Empty;
 
         /// <summary>
-        /// Variant sequence information (required)
+        /// Reference allele (REF) convenience passthrough (null if no VCF).
         /// </summary>
-        public string VariantSequence { get; }
+        public string? ReferenceAllele => VariantCallFormatData?.ReferenceAlleleString;
 
         /// <summary>
-        /// Description of this variation (optional)
+        /// First (primary) alternate allele convenience passthrough if available.
+        /// Returns null if no VCF or ALT not parsable. (Implement inside VariantCallFormat if not already present.)
         /// </summary>
-        public SequenceVariantDescription Description { get; }
+        public string? AlternateAllele => VariantCallFormatData?.AlternateAlleleString; // ensure VariantCallFormat exposes this; if not, remove.
 
         /// <summary>
-        /// Modifications specifically for this variant
+        /// True if this is a point substitution (length 1 → length 1, both non-empty, not a stop).
         /// </summary>
-        public Dictionary<int, List<Modification>> OneBasedModifications { get; }
+        public bool IsPointSubstitution =>
+            OriginalSequence?.Length == 1 &&
+            VariantSequence?.Length == 1 &&
+            VariantSequence != "*" &&
+            OriginalSequence != VariantSequence;
+
+        /// <summary>
+        /// True if substitution length >1 but same length (multi-nucleotide / multi-amino-acid).
+        /// </summary>
+        public bool IsMultiResidueSubstitution =>
+            OriginalSequence?.Length > 1 &&
+            VariantSequence?.Length == OriginalSequence.Length &&
+            OriginalSequence != VariantSequence &&
+            !IsPointSubstitution;
```

Comment:
!IsPointSubstitution is a redundant check as the length check already does this.